src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -243,12 +243,12 @@ public async Task GetAndApplySearchParameterUpdates(CancellationToken cancellati
             if (paramsToAdd.Any())
             {
                 _searchParameterDefinitionManager.AddNewSearchParameters(paramsToAdd);
-
-                // Once added to the definition manager we can update their status
-                await _searchParameterStatusManager.ApplySearchParameterStatus(
-                    updatedSearchParameterStatus.Where(p => p.Status != SearchParameterStatus.Deleted).ToList(),
-                    cancellationToken);
             }
+
+            // Once added to the definition manager we can update their status
+            await _searchParameterStatusManager.ApplySearchParameterStatus(
```

Comment:
>ApplySearchParameterStatus [](http://example.com/codeflow?start=48&length=26)

From reading code it seems like only non deleted status search parameters would be added to `paramsToAdd`.

But it looks like that's not the case?

Can you explain a bit why it's not inside `if (paramsToAdd.Any())`

Reply:
I moved the call to ApplySearchParameterStatus outside the if (paramsToAdd.Any()) because it contains the call to update the _latestSearchParams in SearchParameterStatusManager.  For the case where we detect changes on startup, the paramsToAdd collection will be empty, as those spec defined parameters will not have any SearchParameter resource in the DB.  But, we still need to call ApplySearchParameterStatus.

Reply:
Are we talking about predefined search parameters? 
I just not sure how you can have status and not be in DB...